src/app/modules/submission/submission.utils.ts

```diff
@@ -0,0 +1,30 @@
+import { BasicField, ResponseMode } from '../../../types'
+import { FIELDS_TO_REJECT } from '../../utils/field-validation/config'
+
+type ModeFilterParam = {
+  fieldType: BasicField
+}
+
+export const getModeFilter = (responseMode: ResponseMode) => {
+  switch (responseMode) {
+    case ResponseMode.Email:
+      return emailModeFilter
+    case ResponseMode.Encrypt:
+      return encryptModeFilter
+    default:
+      throw Error('getResponsesForEachField: Invalid response mode parameter')
+  }
+}
+
+const emailModeFilter = <T extends ModeFilterParam>(responses: T[]) => {
+  return responses.filter(
+    ({ fieldType }) => !FIELDS_TO_REJECT.includes(fieldType),
```

Comment:
I'm thinking of not using this import and just forcing a type like in `encryptModeFilter`, but I'm also afraid of splintering the sources of truth. Not a fan of this `FIELDS_TO_REJECT` constant though, KIV.